nexusparser/tools/yaml2nxdl/yaml2nxdl_recursive_build.py

```diff
@@ -103,30 +121,29 @@ def xml_handle_links(obj, k, v):
 
 def recursive_build(obj, dct):
     """
-    obj is the current node where we want to append to, dct is a dictionary object which represents the content of the child in the nested set of dictionaries
+    obj is the current node where we want to append to, 
+    dct is a dictionary object which represents the content of the child in the nested set of dictionaries
     """
     for k, v in iter(dct.items()):
-        print('Processing key '+k+' value v is a dictionary '+str(isinstance(v,dict)))
-        #base class prefix tag removal
-        #k = '(NXentry)'
-        #k = 'sensor_size(NX_INT)'
+        #print('Processing key '+k+' value v is a dictionary '+str(isinstance(v,dict)))
         kName, kType = nx_name_type_resolving(k)
         print('kName: '+kName+' kType: '+kType)
         if kName == '' and kType == '':
             raise ValueError('ERROR: Found an improper YML key !')
-        elif kType.replace('NX','nx_') in nx_base_clss or kType.replace('NX','nx_') in nx_cand_clss:
+        elif kType in nx_base_clss or kType in nx_cand_clss or kType not in nx_type_keys:
             #we can be sure we need to instantiate a new group
             grp = etree.SubElement(obj, 'group')
             if kName != '': #use the custom name for the group
-                grp.set('name', kName) #because we are a base or cand clss kName is a NX<str>
-            #else: #use the NeXus default to infer the name from the base/cand class name stripping the 'NX' decorator prefix
-            #    grp.set('name', kName[2:])
+                grp.set('name', kName) 
+            else: 
+                if kType != '':
+                    grp.set('name', kType)                                  
             grp.set('type', kType)
             if v != None:
```

Comment:
Also not part of PR.

`isinstance(v, dict)` will be false if `v != None`. Also you can combine ifs with and:
```
if isinstance(value, dict) and len(value) > 0:
   build_recursivly(group, value)
```

Btw is should be `v is not None` instead of `v != None`.

Reply:
the _is not_ operator is a better solution only for _None_ or also for _value != {}:_ and for _kName != ''_ ?

Reply:
I actually answered myself trying..